dev/PagerControl/readme.md

```diff
@@ -0,0 +1,75 @@
+# PagerControl
+
+PagerControl allows developers to show items by providing an easy way to handle the user interface of pagination.
+
+## Architecture
+
+The PagerControl is structured into different multiple areas:
+- Handling of control events such as button clicks
+- Property changes
+- Updating NumberPanel buttons
+
+Both the ComboBox and NumberPanel need collections to render that are not easily created by through XAML as they require some logic.
+To pass those collections to the visual layer, we use TemplateSettings. 
+For performance reasons, the collections are stored inside the PagerControl and are not modified through the TemplateSettings object but rather through the reference inside the PagerControl.
+Since the collections on the TemplateSettings are readonly, this way of accessing is always modifying the correct items.
+
+## Template
+
+Below are all the named template parts of the PagerControl template. Next to them is their use case inside the control.
+
+- RootGrid: Referenced to listen to key events to allow accessible keyboarding behavior.
+- FirstPageButton: Referenced to listen to click event and to add the localized UIA name.
+- PreviousPageButton: Referenced to listen to click event and to add the localized UIA name.
+- BoxPanels: Only referenced inside template to quickly hide/show non ButtonPanel controls.
+- NumberBoxDisplay: Referenced to listen to value changed, add localized UIA name and update maximum value.
+- ComboBoxDisplay: Referenced to listen to selected index changed and add localized UIA name.
+- SuffixTextLabel: Only referenced inside the template to hide in infinity mode.
+- TotalNumberOfPagesLabel: Only referenced inside the template to hide in infinity mode.
+- NumberPanelItemsRepeater: Referenced to force layout updating in time and allow correct placement of selection indicator.
+- NumberPanelCurrentPageIndicator: Referenced to update position based on the selected item.
```

Comment:
>NumberPanelCurrentPageIndicator [](start = 2, length = 31)

We probably could do this with a visual state group right? Do you think that would be better? might be a little unwieldy.

Reply:
I am not quite sure if that is possible, after all the template is not really aware of the buttons rendered inside ItemsRepeater. In addition to that, the position for the items might not always be the same (e.g. hidden first page button). Solving this just through visual state groups would probably be very complicated.